sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -141,6 +142,29 @@ case class ShuffleQueryStageExec(
       case _ =>
     }
   }
+
+  private def getPartitionIndexRanges(omittedPartitions: Set[Int]): Array[(Int, Int)] = {
```

Comment:
This method doesn't need to take a parameter. It can access `excludedPartitions` directly.